Slicing issue

Hello,
I've found something weird.

```
objs = Objs.objects.all()
length = len(objs[1:4])
# length equals 3

objs = QuerySetSequence(Objs.objects.all())
length = len(objs[1:4])
# length equals 4
```

Actually, it looks like QuerySetSequence ignores start index and it always equals 0.

As result, it breaks Django pagination. I found a workaround for that - convert QSS to a list, and pass that list to Paginator class. But maybe it should not be like that?

Hey again. Was really busy, but I've got some time today to do some testing. I've made clean install (Python 3.5.2, Django 1.10.6, freshly pip-installed django-querysetsequence). 

Here some funny results:
```
class NoteTests(TestCase):
    def setUp(self):
        # Note model is primitive and contains only one field 'text' VARCHAR(255)
        Note.objects.bulk_create([
            Note(text="Note number 1"),
            Note(text="Note number 2"),
            Note(text="Note number 3"),
            Note(text="Note number 4"),
            Note(text="Note number 5"),
            Note(text="Note number 6")
        ])
    def test_opened_slice(self):
        # This one works
        notes = Note.objects.all()
        qs_notes = notes[1:]
        qss_notes = QuerySetSequence(notes)[1:]

        self.assertEqual(len(qs_notes), len(qss_notes))

    def test_closed_slice_single_qs(self):
        # This one failes
        notes = Note.objects.all()
        qs_notes = notes[1:4]
        qss_notes = QuerySetSequence(notes, notes)[1:4]

        self.assertEqual(len(qs_notes), len(qss_notes))

    def test_closed_slice_two_qs_filter(self):
        # This one works
        notes_1 = Note.objects.filter(pk__lte=2)
        notes_2 = Note.objects.filter(pk__gte=1)

        qss_concat = QuerySetSequence(notes_1, notes_2)

        self.assertNotEqual(len(qss_concat[0:4]), len(qss_concat[2:4]))

    def test_closed_slice_two_qs_all(self):
        # This one fails
        notes_1 = Note.objects.all()
        notes_2 = Note.objects.all()

        qss_concat = QuerySetSequence(notes_1, notes_2)

        self.assertNotEqual(len(qss_concat[0:4]), len(qss_concat[2:4]))
```
Fail messages:
```
======================================================================
FAIL: test_closed_slice_single_qs (modules.homepage.tests.NoteTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pr/modules/homepage/tests.py", line 29, in test_closed_slice_single_qs
    self.assertEqual(len(qs_notes), len(qss_notes))
AssertionError: 2 != 4

======================================================================
FAIL: test_closed_slice_two_qs_all (modules.homepage.tests.NoteTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pr/modules/homepage/tests.py", line 45, in test_closed_slice_two_qs_all
    self.assertNotEqual(len(qss_concat[0:4]), len(qss_concat[2:4]))
AssertionError: 4 == 4

----------------------------------------------------------------------
```

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -3,6 +3,12 @@
 Changelog
 #########
 
+0.7.1 (2017-03-17)
+==================
+
+* [Bugfix] Slicing a ``QuerySetSequence`` did not work properly when the slice
+  reduced the ``QuerySetSequence`` to a single ``QuerySet``. See #23, #24.
+
 0.7 (2016-10-20)
 ================
 

--- a/queryset_sequence/__init__.py
+++ b/queryset_sequence/__init__.py
@@ -224,13 +224,13 @@ class QuerySequence(six.with_metaclass(PartialInheritanceMeta, Query)):
         # from it.
         self.high_mark -= counts[end_index - 1]
 
+        # Some optimization, if there is only one QuerySet, iterate through it.
+        if len(self._querysets) == 1:
+            return iter(self._querysets[0][self.low_mark:self.high_mark])
+
         # Apply the offsets to the edge QuerySets.
         self._querysets[0] = self._querysets[0][self.low_mark:]
         self._querysets[-1] = self._querysets[-1][:self.high_mark]
-
-        # Some optimization, if there is only one QuerySet, iterate through it.
-        if len(self._querysets) == 1:
-            return iter(self._querysets[0])
 
         # For anything left, just chain the QuerySets together.
         return chain(*self._querysets)

--- a/setup.py
+++ b/setup.py
@@ -10,7 +10,7 @@ def long_description():
 setup(
     name='django-querysetsequence',
     packages=find_packages(),
-    version='0.7',
+    version='0.7.1',
     description='Chain together multiple (disparate) QuerySets to treat them as a single QuerySet.',
     long_description=long_description(),
     author='Percipient Networks, LLC',

--- a/tests/test_querysetsequence.py
+++ b/tests/test_querysetsequence.py
@@ -1088,6 +1088,27 @@ class TestSlicing(TestBase):
         self.assertEqual(data[0], 'Biography')
         self.assertEqual(data[1], 'Django Rocks')
 
+    def test_open_slice(self):
+        """Test slicing without an end."""
+        articles = Article.objects.all()
+        qs = articles[1:]
+        qss = QuerySetSequence(articles)[1:]
+
+        self.assertEqual(len(qs), len(qss))
+        self.assertEqual(list(qs), list(qss))
+
+    def test_closed_slice_single_qs(self):
+        """Test slicing if the start and end are within the same QuerySet."""
+        Article.objects.create(title='Another Article', author=self.bob,
+                               publisher=self.mad_magazine)
+
+        articles = Article.objects.all()
+        qs = articles[1:3]
+        qss = QuerySetSequence(articles)[1:3]
+
+        # The length should be the same.
+        self.assertEqual(len(qs), len(qss))
+
 
 class TestGet(TestBase):
     def test_get(self):

